terminus-ui/cohort-date-range/src/cohort-date-range.component.spec.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  Component,
+  Provider,
+  Type, ViewChild,
+} from '@angular/core';
+import {
+  ComponentFixture, fakeAsync, tick,
+} from '@angular/core/testing';
+import { ReactiveFormsModule } from '@angular/forms';
+import {
+  createComponent as createComponentInner, dispatchKeyboardEvent,
+  typeInElement,
+} from '@terminus/ngx-tools/testing';
+import * as testComponents from '@terminus/ui/cohort-date-range/testing';
+// eslint-disable-next-line no-duplicate-imports
+import {
+  getAllDateRangeDebugElements, getRangeInputInstances,
+} from '@terminus/ui/date-range/testing';
+
+import { By } from '@angular/platform-browser';
+import { KEYS } from '@terminus/ngx-tools';
+import {
+  TsDateRangeComponent,
+  TsDateRangeModule,
+} from '@terminus/ui/date-range';
+import {
+  createKeydownEvent, getSelectElement, getSelectInstance,
+  getSelectTriggerElement,
+} from '@terminus/ui/select/testing';
+import {
+  TsCohortDateRangeComponent, TsCohortDateRangeModule,
+} from './cohort-date-range.module';
+
+
+describe(`TsCohortDateRangeComponent`, () => {
+  let fixture;
+  let hostInstance;
+  let startInputInstance;
+  let formFieldElement;
+  let testComponent: testComponents.Basic;
+  let cohortDebugElement;
+  let cohortInstance;
+
+  function setupComponent(exampleComponent) {
+    fixture = createComponent(testComponents[exampleComponent]);
+    fixture.detectChanges();
+    hostInstance = fixture.componentInstance;
+    testComponent = fixture.debugElement.componentInstance;
+    fixture.detectChanges();
+    startInputInstance = getRangeInputInstances(fixture)[0];
+    formFieldElement = fixture.debugElement.query(By.css('.ts-form-field'));
+    cohortDebugElement = fixture.debugElement.query(By.directive(TsCohortDateRangeComponent));
+    cohortInstance = cohortDebugElement.componentInstance;
+  }
+
+  describe(`allowCustomsDate`, () => {
+    beforeEach(() => {
+      setupComponent('Basic');
+    });
+    test(`should disable date range if allowCustomsDate is false`, () => {
```

Comment:
The ticket notes that when allow custom is false, the date range should be read only rather than disabled.